### PR TITLE
Always use current profile on thread events

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/23648

Events in a thread always have to display the current profile image as there's no state events in the thread timeline. This ensures that we always get the member from the room state, and that fixes all the issues with missing senders and displaying the fallback user id

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Always use current profile on thread events ([\#9524](https://github.com/matrix-org/matrix-react-sdk/pull/9524)). Fixes vector-im/element-web#23648.

### CODE REVIEW

File: src/components/views/avatars/MemberAvatar.tsx [hunk 1]
```diff
@@ -46,100 +47,85 @@ interface IProps extends Omit<React.ComponentProps<typeof BaseAvatar>, "name" |
...
+    const cli = useContext(MatrixClientContext);
+    const card = useContext(CardContext);
+    const roomContext = useContext(RoomContext);
+
+    const [name, setName] = useState<string | undefined>();
+    const [title, setTitle] = useState<string | undefined>();
+    const [imageUrl, setImageUrl] = useState<string | undefined>();
+    const [userId, setUserId] = useState<string | undefined>();
+
+    useEffect(() => {
```
Comment (florianduros) [inline src/components/views/avatars/MemberAvatar.tsx:66]: I suggest to create a custom hook here to improve the readability of the component

File: src/components/views/avatars/MemberAvatar.tsx [hunk 2]
```diff
@@ -46,100 +47,85 @@ interface IProps extends Omit<React.ComponentProps<typeof BaseAvatar>, "name" |
...
+        const useOnlyCurrentProfiles = (member && !props.forceHistorical
+            && SettingsStore.getValue("useOnlyCurrentProfiles"))
+            || roomContext?.timelineRenderingType === TimelineRenderingType.ThreadsList
+            || roomContext?.timelineRenderingType === TimelineRenderingType.Thread;
 
-    private static getState(props: IProps): IState {
-        let member = props.member;
-        if (member && !props.forceHistorical && SettingsStore.getValue("useOnlyCurrentProfiles")) {
-            const room = MatrixClientPeg.get().getRoom(member.roomId);
+        if (useOnlyCurrentProfiles) {
```
Comment (florianduros) [inline src/components/views/avatars/MemberAvatar.tsx:74]: This code is duplicated in the component below, maybe we can split this hook in multiple hook and derived state:

```ts
function getMember(...) {
   if (SettingsStore.getValue("useOnlyCurrentProfiles")
        || roomContext.timelineRenderingType === TimelineRenderingType.ThreadsList
        || roomContext.timelineRenderingType === TimelineRenderingType.Thread
    ) {
        const room = cli.getRoom(mxEvent.getRoomId());
        if (room) {
            member = room.getMember(mxEvent.getSender());
        }
    return member
}

function MemberAvatar() {
   const member = props.member && !props.forceHistorical && getMember()
   // maybe use memo to only compute when needed
   const name = member?.name || props.fallbackUserId || ''
  // maybe use memo to only compute when needed
   const userTitle = UserIdentifierCustomisations.getDisplayUserIdentifier(
                member.userId, { roomId: member?.roomId },
            );
   // maybe use memo to only compute when needed
   const title = member?.name ? props.title || userTitle :  props.fallbackUserId || ''
   const userId = member?.userId ?? props.fallbackUserId
   // maybe use memo to only compute when needed
   const imageUrl = member?.name && member.getMxcAvatarUrl() && mediaFromMxc(...)

   ...
}
```

**Review by florianduros — CHANGES_REQUESTED** [review]

File: src/hooks/room/useRoomMemberProfile.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    member: propMember,
+    forceHistorical = false,
+}: {
+    userId: string | undefined;
+    member?: RoomMember;
+    forceHistorical?: boolean;
+}): RoomMember {
+    const [member, setMember] = useState<RoomMember | undefined>(propMember);
+
+    const context = useContext(RoomContext);
```
Comment (florianduros) [inline src/hooks/room/useRoomMemberProfile.ts:34]: You can also use `useRoomContext()` here but just a question of preference :)

**Review by florianduros — APPROVED** [review]
LGFM

File: src/hooks/room/useRoomMemberProfile.ts (hunk 3 above)
Comment (gsouquet) [reply src/hooks/room/useRoomMemberProfile.ts:34]: Uhh TIL !

Comment (florianduros) [reply src/hooks/room/useRoomMemberProfile.ts:34]: I added it few weeks ago ^^

**Pull request merged**